sdk/storage/storage-blob/src/Pipeline.ts

```diff
@@ -59,177 +53,42 @@ export {
-
-/**
- * Option interface for newPipeline() method.
- *
- * @export
- * @interface StoragePipelineOptions
- */
-export interface StoragePipelineOptions {
-  proxy?: ProxySettings | string;
-  /**
-   * Telemetry configures the built-in telemetry policy behavior.
-   *
-   * @type {TelemetryOptions}
-   * @memberof StoragePipelineOptions
-   */
-  telemetry?: TelemetryOptions;
-  /**
-   * Configures the built-in retry policy behavior.
-   *
-   * @type {RetryOptions}
-   * @memberof StoragePipelineOptions
-   */
-  retryOptions?: RetryOptions;
-  /**
-   * Keep alive configurations. Default keep-alive is enabled.
-   *
-   * @type {KeepAliveOptions}
-   * @memberof StoragePipelineOptions
-   */
-  keepAliveOptions?: KeepAliveOptions;
-
-  /**
-   * Configures the HTTP pipeline logger.
-   *
-   * @type {IHttpPipelineLogger}
-   * @memberof StoragePipelineOptions
-   */
-  logger?: IHttpPipelineLogger;
-  /**
-   * Configures the HTTP client to send requests and receive responses.
-   *
-   * @type {IHttpClient}
-   * @memberof StoragePipelineOptions
-   */
-  httpClient?: IHttpClient;
-}
-
-/**
- * Creates a new Pipeline object with Credential provided.
+ * Creates a ServiceClientOptions object with Credential provided.
  *
  * @export
  * @param {SharedKeyCredential | AnonymousCredential | TokenCredential} credential Such as AnonymousCredential, SharedKeyCredential
  *                                                  or a TokenCredential from @azure/identity.
- * @param {StoragePipelineOptions} [pipelineOptions] Optional. Options.
- * @returns {Pipeline} A new Pipeline object.
+ * @param {PipelineOptions} [pipelineOptions] Optional. Options.
+ * @returns {ServiceClientOptions} ServiceClientOptions for use with @azure/core-http
  */
 export function newPipeline(
   credential: SharedKeyCredential | AnonymousCredential | TokenCredential,
-  pipelineOptions: StoragePipelineOptions = {}
-): Pipeline {
-  // Order is important. Closer to the API at the top & closer to the network at the bottom.
-  // The credential's policy factory must appear close to the wire so it can sign any
-  // changes made by other factories (like UniqueRequestIDPolicyFactory)
-  const factories: RequestPolicyFactory[] = [
-    tracingPolicy(),
-    new KeepAlivePolicyFactory(pipelineOptions.keepAliveOptions),
-    new TelemetryPolicyFactory(pipelineOptions.telemetry),
-    new UniqueRequestIDPolicyFactory(),
-    new BrowserPolicyFactory(),
-    deserializationPolicy(), // Default deserializationPolicy is provided by protocol layer
-    new RetryPolicyFactory(pipelineOptions.retryOptions),
-    logPolicy({
+  pipelineOptions: PipelineOptions = {}
+): ServiceClientOptions {
+  const userAgentParts: string[] = [];
+  if (isNode) {
+    if (pipelineOptions.userAgentOptions && pipelineOptions.userAgentOptions.userAgentPrefix) {
+      userAgentParts.push(pipelineOptions.userAgentOptions.userAgentPrefix);
+    }
+    userAgentParts.push(`azsdk-js-storageblob/${SDK_VERSION}`);
```

Comment:
Will NODE-VERSION part be assembled properly? e.g. (NODE-VERSION 4.9.1; Windows_NT 10.0.16299)

Reply:
This is generated here, and I think it's equivalent?: https://github.com/Azure/azure-sdk-for-js/blob/f2940c39d5053071059071f024bd728193219689/sdk/core/core-http/lib/policies/msRestUserAgentPolicy.ts#L12